engine/sound/src/sound.cpp

```diff
@@ -518,25 +518,49 @@ namespace dmSound
         sd->m_Index = index;
         sd->m_Data = 0;
         sd->m_Size = 0;
-        sd->m_DataCallbacks.m_Context = 0;
-        sd->m_DataCallbacks.m_GetData = 0;
+        sd->m_DataCallbacks.m_Context = cbk_ctx;
+        sd->m_DataCallbacks.m_GetData = cbk;
         sd->m_RefCount = 1;
 
-        Result result = SetSoundDataNoLock(sd, sound_buffer, sound_buffer_size);
-        if (result == RESULT_OK)
+        Result result = RESULT_OK;
+        if (sound_buffer != 0)
+        {
+            result = SetSoundDataNoLock(sd, sound_buffer, sound_buffer_size);
+        }
+        if (RESULT_OK == result)
             *sound_data = sd;
         else
             DeleteSoundData(sd);
-
         return result;
     }
 
+    Result NewSoundData(const void* sound_buffer, uint32_t sound_buffer_size, SoundDataType type, HSoundData* sound_data, dmhash_t name)
+    {
+        assert(sound_buffer);
+        return NewSoundDataInternal(sound_buffer, sound_buffer_size, 0, 0, type, sound_data, name);
```

Comment:
To make internal api more similar, we could use an (internal) function callback for the reading as well. (maybe later)